src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/profiler/ProfilerEditingTarget.java

```diff
@@ -671,16 +696,61 @@ public void execute()
       };
    }
    
-   private static void onMessage(String message, String file, int line)
-   {  
+   private void onMessage(final String message,
+                          final String file,
+                          final String details,
+                          final int line)
+   {
       if (message == "sourcefile")
       {
-         FilePosition filePosition = FilePosition.create(line, 0);
-         CodeNavigationTarget navigationTarget = new CodeNavigationTarget(file, filePosition);
-         
-         RStudioGinjector.INSTANCE.getFileTypeRegistry().editFile(
-               FileSystemItem.createFile(navigationTarget.getFile()),
-               filePosition);
+         server_.profileSources(file, new ServerRequestCallback<String>()
```

Comment:
It looks like this RPC doesn't do more than check to see whether a file exists. There's already an RPC that does this -- use `FilesServerOperations.stat`; it returns a `FileSystemItem` on which you can call `exists()`j.